Namespace compiler: UnicodeDecodeError: 'gbk' codec can't decode byte 0x92

Hi
        Can anyone help me? Thanks very much!
        I follow the step in open62541 documentation"https://open62541.org/doc/current/building.html#building-with-cmake-on-windows".
        The git(download) path is "https://github.com/open62541/open62541.git".
        I have installed the "Pythone Six" modual, I have builded it at Windows 7(Visual studio 2015 14 or MinGW_Makefiles both fail) and Ubuntu 16.04.03. The Error is like follow:

[  1%] Built target open62541-generator-statuscode
[  3%] Built target open62541-generator-types
[  3%] Generating src_generated/ua_namespace0.c, src_generated/ua_namespace0.h
INFO:__main__:Preprocessing E:/Professional/Cross Compiling/Multi-Platform Lib/open62541-master/tools/schema/Opc.Ua.NodeSet2.Minimal.xml
Traceback (most recent call last):
  File "E:/Professional/Cross Compiling/Multi-Platform Lib/open62541-master/tools/nodeset_compiler/nodeset_compiler.py", line 138, in <module>
    ns.addNodeSet(xmlfile, typesArray=getTypesArray(nsCount))
  File "E:\Professional\Cross Compiling\Multi-Platform Lib\open62541-master\tools\nodeset_compiler\nodeset.py", line 217, in addNodeSet
    orig_namespaces = extractNamespaces(xmlfile)  # List of namespaces used in the xml file
  File "E:\Professional\Cross Compiling\Multi-Platform Lib\open62541-master\tools\nodeset_compiler\nodeset.py", line 59, in extractNamespaces
    line = infile.readline()
UnicodeDecodeError: 'gbk' codec can't decode byte 0x92 in position 5077: illegal multibyte sequence
CMakeFiles\open62541-generator-namespace.dir\build.make:67: recipe for target 'src_generated/ua_namespace0.c' failed
mingw32-make[2]: *** [src_generated/ua_namespace0.c] Error 1
CMakeFiles\Makefile2:347: recipe for target 'CMakeFiles/open62541-generator-namespace.dir/all' failed
mingw32-make[1]: *** [CMakeFiles/open62541-generator-namespace.dir/all] Error 2
Makefile:150: recipe for target 'all' failed
mingw32-make: *** [all] Error 2

 - [Latest commit 9aa0b24 ] open62541 Version (release number or git tag):
 - [No ] Other OPC UA SDKs used (client or server): 
 - [ Windows7 or Ubuntu 16.04.3] Operating system:
 - [ ] Logs (with `UA_LOGLEVEL` set as low as necessary) attached
 - [ ] Wireshark network dump attached
 - [ ] Self-contained code example attached
 - [ ] Critical issue

--- a/tools/nodeset_compiler/backend_graphviz.py
+++ b/tools/nodeset_compiler/backend_graphviz.py
@@ -1,5 +1,6 @@
 from nodeset import *
 import graphviz as gv
+import codecs
 
 def nodePrintDot(node):
     cleanname = "node_" + str(node.id).replace(";", "").replace("=", "")
@@ -39,7 +40,7 @@ def printDotGraphWalk(nodeset, depth=1, filename="out.dot", rootNode=None,
     else:
         root = rootNode
 
-    file = open(filename, 'w+')
+    file = codecs.open(filename, 'w+', encoding='utf-8')
 
     if root == None:
         return

--- a/tools/nodeset_compiler/backend_open62541.py
+++ b/tools/nodeset_compiler/backend_open62541.py
@@ -23,6 +23,7 @@ import string
 from collections import deque
 from os.path import basename
 import logging
+import codecs
 try:
     from StringIO import StringIO
 except ImportError:
@@ -162,7 +163,7 @@ def reorderNodesMinDependencies(nodeset):
 def generateOpen62541Code(nodeset, outfilename, supressGenerationOfAttribute=[], generate_ns0=False, internal_headers=False, typesArray=[], max_string_length=0):
     outfilebase = basename(outfilename)
     # Printing functions
-    outfileh = open(outfilename + ".h", r"w+")
+    outfileh = codecs.open(outfilename + ".h", r"w+", encoding='utf-8')
     outfilec = StringIO()
 
     def writeh(line):
@@ -252,6 +253,6 @@ UA_StatusCode retVal = UA_STATUSCODE_GOOD;
     fullCode = outfilec.getvalue()
     outfilec.close()
 
-    outfilec = open(outfilename + ".c", r"w+")
+    outfilec = codecs.open(outfilename + ".c", r"w+", encoding='utf-8')
     outfilec.write(fullCode)
     outfilec.close()

--- a/tools/nodeset_compiler/nodeset.py
+++ b/tools/nodeset_compiler/nodeset.py
@@ -21,7 +21,8 @@ import sys
 import xml.dom.minidom as dom
 from struct import pack as structpack
 from time import struct_time, strftime, strptime, mktime
-import logging;
+import logging
+import codecs
 
 logger = logging.getLogger(__name__)
 
@@ -53,7 +54,7 @@ def extractNamespaces(xmlfile):
     # attempts to do just that.
 
     namespaces = ["http://opcfoundation.org/UA/"]
-    infile = open(xmlfile.name)
+    infile = codecs.open(xmlfile.name, encoding='utf-8')
     foundURIs = False
     nsline = ""
     line = infile.readline()

--- a/tools/nodeset_compiler/nodeset_testing.py
+++ b/tools/nodeset_compiler/nodeset_testing.py
@@ -1,5 +1,6 @@
 #!/usr/bin/env python
 
+import codecs
 from nodeset import *
 
 class testing:
@@ -23,7 +24,7 @@ class testing:
         self.ns.allocateVariables()
 
         bin = self.ns.buildBinary()
-        f = open("binary.base64", "w+")
+        f = codecs.open("binary.base64", "w+", encoding='utf-8')
         f.write(bin.encode("base64"))
         f.close()
 
@@ -65,7 +66,7 @@ class testing_open62541_header:
         logger.debug("Phase 3: Calling C Printers")
         code = self.ns.printOpen62541Header()
 
-        codeout = open("./open62541_nodeset.c", "w+")
+        codeout = codecs.open("./open62541_nodeset.c", "w+", encoding='utf-8')
         for line in code:
             codeout.write(line + "\n")
         codeout.close()

